### PR TITLE
KAFKA-16246: Cleanups in ConsoleConsumer

### PR DESCRIPTION
**Pull request by wernerdv**
Removed Optional where it makes sense and removed the argument checking logic that duplicates existing in ConsoleConsumerOptions#checkRequiredArgs().

Also fixed NPE when the `--topic` option is not used (because uses `Optional.of(opts.topicArg())` instead of `Optional.ofNullable`).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wernerdv** [comment]
Hello, @mimaison @OmniaGM
Please, take a look.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java [hunk 1]
```diff
@@ -148,43 +143,26 @@ static boolean checkErr(PrintStream output) {
...
+            timeoutMs = opts.timeoutMs() >= 0 ? opts.timeoutMs() : Long.MAX_VALUE;
+
+            if (topic.isPresent()) {
+                if (opts.partitionArg().isPresent()) {
+                    seek(topic.get(), opts.partitionArg().getAsInt(), opts.offsetArg());
+                } else {
+                    consumer.subscribe(Collections.singletonList(topic.get()));
+                }
             } else {
-                throw new IllegalArgumentException("An invalid combination of arguments is provided. " +
```
Comment (OmniaGM) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java:184]: We shouldn't delete this instead it needs to be moved to the validation of `ConsoleConsumerOptions` as the args are not a valid combination. Do we have an alternative exception to this or are we okay with changing the error of this case?

**Review by OmniaGM — COMMENTED** [review]
Thanks for the pr, nice cleanup! I had a quick look and left one comment

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java (hunk 1 above)
Comment (OmniaGM) [reply tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java:184]: I just noticed your note in the pr description. We don't have the exact same error message in `ConsoleConsumerOptions#checkRequiredArgs()`. I think it's okay to delete this error message from here if `ConsoleConsumerOptions#checkRequiredArgs()` protect us from initialise this `ConsumerWrapper` with invalid combination

Comment (wernerdv) [reply tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java:184]: Thank you for the review.
It seems to me that all the checks that were previously here are already implemented in `ConsoleConsumerOptions#checkRequiredArgs()`, namely:
- Exactly one of --include/--topic is required.
- The topic is required when partition is specified.
- The partition is required when offset is specified.

This is protect us from initialise this ConsumerWrapper with invalid combination. 
I couldn't find a case where I would catch an IllegalArgumentException with the current message.
Please correct me if I'm wrong.

**Review by OmniaGM — APPROVED** [review]

**Comment by wernerdv** [comment]
Hello, @mimaison
Please, take a look.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java [hunk 2]
```diff
@@ -148,43 +143,26 @@ static boolean checkErr(PrintStream output) {
...
-            if (topic.isPresent() && partitionId.isPresent() && offset.isPresent() && !includedTopics.isPresent()) {
-                seek(topic.get(), partitionId.getAsInt(), offset.getAsLong());
-            } else if (topic.isPresent() && partitionId.isPresent() && !offset.isPresent() && !includedTopics.isPresent()) {
-                // default to latest if no offset is provided
-                seek(topic.get(), partitionId.getAsInt(), ListOffsetsRequest.LATEST_TIMESTAMP);
-            } else if (topic.isPresent() && !partitionId.isPresent() && !offset.isPresent() && !includedTopics.isPresent()) {
-                consumer.subscribe(Collections.singletonList(topic.get()));
-            } else if (!topic.isPresent() && !partitionId.isPresent() && !offset.isPresent() && includedTopics.isPresent()) {
-                consumer.subscribe(Pattern.compile(includedTopics.get()));
+            timeoutMs = opts.timeoutMs() >= 0 ? opts.timeoutMs() : Long.MAX_VALUE;
```
Comment (mimaison) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java]: Could we merge this with the `timeoutMs()` method from `ConsoleConsumerOptions`?

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java [hunk 3]
```diff
@@ -148,43 +143,26 @@ static boolean checkErr(PrintStream output) {
...
-                               OptionalLong offset,
-                               Optional<String> includedTopics,
-                               Consumer<byte[], byte[]> consumer,
-                               long timeoutMs) {
-            this.topic = topic;
-            this.partitionId = partitionId;
-            this.offset = offset;
-            this.includedTopics = includedTopics;
+        public ConsumerWrapper(ConsoleConsumerOptions opts, Consumer<byte[], byte[]> consumer) {
+            Optional<String> topic = Optional.ofNullable(opts.topicArg());
```
Comment (mimaison) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java]: Could `topicArg()` directly return `Optional<String>`?
Same below for `includedTopicsArg()`.

**Review by mimaison — COMMENTED** [review]
Thanks for the PR! I left a couple of questions.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java (hunk 2 above)
Comment (wernerdv) [reply tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java]: Done.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java (hunk 3 above)
Comment (wernerdv) [reply tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumer.java]: Done.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java [hunk 4]
```diff
@@ -322,6 +324,20 @@ private void invalidOffset(String offset) {
                 "'earliest', 'latest', or a non-negative long.");
     }
 
+    private long parseTimeoutMs() {
+        long timeout;
+        if (options.has(timeoutMsOpt)) {
+            timeout = options.valueOf(timeoutMsOpt);
+            if (timeout < 0) {
+                CommandLineUtils.printUsageAndExit(parser, "The provided timeout-ms value '" + timeout +
```
Comment (mimaison) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java]: It seems previously we wouldn't fail if a negative value was provided.

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java [hunk 5]
```diff
@@ -385,10 +404,14 @@ String bootstrapServer() {
...
-                ? options.valueOf(includeOpt)
-                : options.valueOf(whitelistOpt);
+    Optional<String> includedTopicsArg() {
+        if (options.has(includeOpt)) {
+            return Optional.of(options.valueOf(includeOpt));
+        } else if (options.has(whitelistOpt)) {
+            return Optional.of(options.valueOf(whitelistOpt));
+        } else {
+            return Optional.empty();
+        }
```
Comment (mimaison) [inline tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java]: Could this be simplified into:
```
return options.has(includeOpt)
                ? Optional.of(options.valueOf(includeOpt))
                : Optional.ofNullable(options.valueOf(whitelistOpt));
```

**Review by mimaison — COMMENTED** [review]
Thanks for the updates! I have a few more questions

File: tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java (hunk 4 above)
Comment (wernerdv) [reply tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java]: Returned to the previous logic.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR!

**Pull request merged**